tuf/api/exceptions.py

```diff
@@ -29,6 +29,10 @@ class BadVersionNumberError(RepositoryError):
     """An error for metadata that contains an invalid version number."""
 
 
+class VerifiedVersionNumberError(RepositoryError):
+    """An error for metadata containing a previously verified version number."""
```

Comment:
Hmm, the error is not that it's a _verified_ number, the error is that it's equal to current version in a case where equal version is not allowed... so `EqualVersionNumberError` maybe. I think it makes sense to derive from BadVersionNumberError (the error is in practice purely internal but I think this would be clearer to users reading exceptions.py)

Reply:
I used `Verified` to underline that this timestamp has been already verified and that's why it doesn't need a new version.
I don't have strong preferences and will change the error name.